lib/sensors/bms.cpp

```diff
@@ -0,0 +1,46 @@
+#include "bms.hpp"
+
+namespace hyped::sensors {
+Bms::Bms(core::ILogger &logger, io::HardwareCan &can) : logger_(logger), can_(can)
+{
+}
+
+std::optional<BmsData> Bms::receiveBmsData()
+{
+  // TODO: Implement
+  // Outline of possible implementation:
+  BmsData bms_data;
+  const auto optional_cell_temperatures = receiveCellTemperature();
+  if (!optional_cell_temperatures) {
+    logger_.log(core::LogLevel::kFatal, "failed to receive cell temperatures");
+  }
+  bms_data.temperatures = optional_cell_temperatures.value();
+  // return bms_data;
+  return std::nullopt;
+}
+
+std::optional<std::array<core::Float, NUM_BMS_CELLS>> Bms::receiveCellTemperature()
+{
+  // TODO: Implement
+  // Outline of possible implementation:
+  const auto bmsProcessor = std::make_shared<BmsProcessor>(logger_);
+  can_.addProcessor(static_cast<std::uint16_t>(BmsPacketIds::kBmsStatusTemperatures), bmsProcessor);
+  const auto frame = can_.receive();
+  if (!frame) { logger_.log(core::LogLevel::kFatal, "Failed to receive cell temperatures"); }
+  // would then parse frame and return temperatures
+  return std::nullopt;
```

Comment:
for reference (i think you know this already but will put it here just in case):
`can_.addProcessor` should be called on initialisation, then when a message with the ID comes through `can_ ` will call `BmsProcessor.processMessage(received_frame)`. This is the function that should then call things like `updateCellTemperatures` for example, which can deal with the actual contents of the message